src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,267 @@
+                preparedStatement.setLong(1, id);
+                ResultSet resultSet = preparedStatement.executeQuery();
+                Car car = null;
+                if (resultSet.next()) {
+                    car = getCar(resultSet);
+                    car.setDrivers(geListDriver(connection, car.getId()));
+                }
+                return Optional.ofNullable(car);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error retrieving car " + "by id = " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query
+                = "SELECT * FROM cars AS c "
+                + "LEFT JOIN manufacturers AS m ON "
+                + "c.manufacturers_id = m.id "
+                + "WHERE c.is_deleted = FALSE";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            try (PreparedStatement preparedStatement
+                         = connection.prepareStatement(query)) {
+                ResultSet resultSet = preparedStatement.executeQuery();
+                Car car;
+                while (resultSet.next()) {
+                    car = getCar(resultSet);
+                    car.setDrivers(geListDriver(connection, car.getId()));
+                    cars.add(car);
+                }
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error retrieving cars", e);
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateQuery
+                = "UPDATE cars SET model = ?, manufacturers_id = ? WHERE id = ?";
+        String deleteQuery
+                = "DELETE FROM cars_drivers WHERE car_id = ?";
+        String addQuery
+                = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCarStatement = connection.prepareStatement(
+                        updateQuery);
+                PreparedStatement deleteDriversStatement = connection.prepareStatement(
+                        deleteQuery);
+                PreparedStatement addDriverStatement = connection.prepareStatement(
+                        addQuery)) {
+
+            updateCarStatement.setString(1, car.getModel());
+            updateCarStatement.setLong(2, car.getManufacturer().getId());
+            updateCarStatement.setLong(3, car.getId());
+            updateCarStatement.executeUpdate();
+
+            deleteDriversStatement.setLong(1, car.getId());
+            deleteDriversStatement.executeUpdate();
+
+            for (Driver driver : car.getDrivers()) {
+                addDriverStatement.setLong(1, car.getId());
+                addDriverStatement.setLong(2, driver.getId());
+                addDriverStatement.executeUpdate();
+            }
```

Comment:
better add private methods `deleteDriversFromCar(Long carId, Connection connection)` and `addDriversToCar(Car car, Connection connection)`